### DISCUSSION TITLE
Gift Card application + subscription products in Cart and Checkout

### DISCUSSION THREAD

**Discussion by matthewlein** [Shopify/storefront-api-feedback#162] [Help]
I am wondering why carts and checkouts don't have (near) mutation parity?

I have an application that uses the storefront API as a headless store.

I would like something that looks like a cart to users.

1. Add products, checkout, the usual cart stuff
2. It needs to be able to apply gift cards.
3. I would like to be able to add subscription products, like a yearly recurring membership.

Currently I use Checkouts. I create a Checkout for each user to act as a cart. They add products, apply gift cards, discount codes, etc. It works well. However, I don't think there is a way to add subscription products to a Checkout.
 
The alternative would be to use Carts. They can add subscription products. However, I think there is no way to apply gift cards to a Cart. I'm not sure what other limitations there would be.
 
If Carts had all the mutations/features of Checkouts (applying gift cards and more), I could use them.
Or, if Checkouts line items accepted a subscription product (through seller id or whatever method makes that work) I could use them.

But as is now, I don't seem to be able to use either for this purpose.

**Comment by lukasgjetting** [comment]
[J-ROM's comment](https://community.shopify.com/c/subscriptions/subscriptions-on-checkout-line-items-or-gift-card-apply-on-cart/m-p/1866606/highlight/true#M1553) from back in 2022 indicates that there are no plans to support Selling Plans via the Checkout API.

Would be interested in hearing @cocoahero if there are any plans to support applying gift cards via the Cart API?

A mutation similar to [checkoutGiftCardsAppend](https://shopify.dev/docs/api/storefront/2023-04/mutations/checkoutGiftCardsAppend) but for the Cart API.

**Reply by vixdug** [reply]
Hi @lukasgjetting - adding support to append a Gift Cart to a cart is now on the roadmap. I will update here when this functionality is available in unstable to test.

**Reply by lukasgjetting** [reply]
Hi Victoria - that's awesome, thanks for updating! Looking forward to it. :-)
